Source/WebKit/UIProcess/API/Swift/WebView.swift

```diff
@@ -140,4 +157,67 @@ extension WebViewRepresentable: NSViewRepresentable {
 }
 #endif
 
+@MainActor
+private final class Coordinator {
+    init(cameraCaptureState: Binding<WKMediaCaptureState>?, microphoneCaptureState: Binding<WKMediaCaptureState>?) {
+        self.cameraCaptureState = cameraCaptureState
+        self.microphoneCaptureState = microphoneCaptureState
+    }
+
+    deinit {
+        for observation in observations {
+            observation.invalidate()
+        }
+    }
+
+    private var cameraCaptureState: Binding<WKMediaCaptureState>?
+    private var microphoneCaptureState: Binding<WKMediaCaptureState>?
+
+    private var observations: Set<NSKeyValueObservation> = []
+
+    weak var owner: WebPage_v0? = nil {
+        didSet {
+            updateObservations()
+        }
+    }
+
+    private func createObservation<Value>(on webView: WKWebView, keyPath: KeyPath<Coordinator, Binding<Value>?>, backedBy backingKeyPath: KeyPath<WKWebView, Value>) -> NSKeyValueObservation where Value: Equatable {
```

Comment:
Might not hurt to line break since it’s got a lot going on.
```suggestion
    private func createObservation<Value>(
        on webView: WKWebView,
        keyPath: KeyPath<Coordinator, Binding<Value>?>,
        backedBy backingKeyPath: KeyPath<WKWebView, Value>
    ) -> NSKeyValueObservation where Value: Equatable {
```